### PR TITLE
Search games endpoints

### PR DESCRIPTION
**Pull request by PedroOliveiraPT**

### CODE REVIEW

**Review by HerouFenix — CHANGES_REQUESTED** [review]
Code smells and pipeline failed

File: api/src/main/java/com/api/demo/grid/controller/GridRestController.java [hunk 1]
```diff
@@ -1,4 +1,69 @@
 package com.api.demo.grid.controller;
+import com.api.demo.grid.models.Game;
+import com.api.demo.grid.service.GridService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
```
Comment (oEscal) [inline api/src/main/java/com/api/demo/grid/controller/GridRestController.java]: atenção ao import *

File: api/src/test/java/com/api/demo/grid/controller/GridRestControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.hamcrest.Matchers.*;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/controller/GridRestControllerTest.java]: atenção ao import *

File: api/src/main/java/com/api/demo/grid/controller/GridRestController.java [hunk 3]
```diff
@@ -1,4 +1,69 @@
...
 
+import java.util.List;
+
+@RestController
+@RequestMapping("/grid")
 public class GridRestController {
+    private static final String ERROR = "No Game found with Id ";
+
+    @Autowired
+    private GridService gridService;
```
Comment (oEscal) [inline api/src/main/java/com/api/demo/grid/controller/GridRestController.java:21]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/DemoApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+class DemoApplicationTest {
+    @Autowired
+    private GridRestController controller;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/DemoApplicationTest.java:13]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/controller/GridRestControllerIT.java [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@AutoConfigureMockMvc
+@AutoConfigureTestDatabase
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = DemoApplication.class)
+class GridRestControllerIT {
+    @Autowired
+    private MockMvc mockMvc;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/controller/GridRestControllerIT.java]: tem em atenção as naming conventions (nesta e em todas desta classe): [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/controller/GridRestControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.Arrays;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@AutoConfigureTestDatabase
+class GridRestControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/controller/GridRestControllerTest.java]: tem em atenção as naming conventions (em toda a classe): [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/repository/DeveloperRepositoryTest.java [hunk 7]
```diff
@@ -20,7 +20,7 @@
     private DeveloperRepository repository;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/repository/DeveloperRepositoryTest.java]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/repository/GameGenreRepositoryTest.java [hunk 8]
```diff
@@ -19,7 +19,7 @@
     private GameGenreRepository repository;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/repository/GameGenreRepositoryTest.java]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/repository/GameRepositoryTest.java [hunk 9]
```diff
@@ -29,35 +26,35 @@
     private GameRepository repository;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/repository/GameRepositoryTest.java]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/repository/PublisherRepositoryTest.java [hunk 10]
```diff
@@ -21,7 +21,7 @@
     private PublisherRepository repository;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/repository/PublisherRepositoryTest.java]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

File: api/src/test/java/com/api/demo/grid/service/GridServiceTest.java [hunk 11]
```diff
@@ -44,7 +44,7 @@
     private Publisher publisher;
```
Comment (oEscal) [inline api/src/test/java/com/api/demo/grid/service/GridServiceTest.java:44]: tem em atenção as naming conventions: [link](https://source.android.com/setup/contribute/code-style#follow-field-naming-conventions)

**Review by oEscal — CHANGES_REQUESTED** [review]
para além disso, muda maybe o nome do controller para o que é que o códio nele faz mesmo, acho que fica mais organizado

**Pull request merged**